beta-src/src/components/ui/WDMoveControls.tsx

```diff
@@ -30,6 +41,59 @@ const WDMoveControls: React.FC<WDMoveControlsProps> = function ({
       isMobile = false;
       break;
   }
+
+  const click = (type: Move) => {
+    dispatch(
+      gameApiSliceActions.processMapClick({
+        name: undefined,
+        clickObject: "save_button",
+      }),
+    );
+    if ("currentOrders" in data && "contextVars" in data) {
+      const { currentOrders, contextVars } = data;
+      if (contextVars && currentOrders) {
+        const orderUpdates: IOrderData[] = [];
+        currentOrders.forEach((o) => {
+          const updateReference = ordersMeta[o.id].update;
+          let orderUpdate: IOrderData = o;
+          if (updateReference) {
+            orderUpdate = {
+              ...o,
+              ...{
+                type: updateReference.type,
+                toTerrID: updateReference.toTerrID,
+              },
+            };
+          }
+          orderUpdates.push(orderUpdate);
+        });
+        const orderSubmission = {
+          orderUpdates,
+          context: contextVars.context,
+          contextKey: contextVars.contextKey,
+        };
+        dispatch(saveOrders(orderSubmission));
+      }
+    }
+    if (type === Move.READY) {
+      toggleState(type);
+    }
+  };
+
+  const ordersMetaValues = Object.values(ordersMeta);
+  const ordersLength = ordersMetaValues.length;
+  const ordersSaved = ordersMetaValues.reduce(
+    (acc, meta) => acc + +meta.saved,
+    0,
+  );
+
+  if (
+    (ordersLength === ordersSaved && save) ||
+    (ordersLength !== ordersSaved && !save)
```

Comment:
is this check even needed seems like we to save for either case.

Reply:
It is needed otherwise the component will try to re-render indefinitely since save will always be toggled.